pkg/defaults/catsrcHelpers.go

```diff
@@ -139,8 +139,13 @@ func AreCatsrcSpecsEqual(spec1 *olm.CatalogSourceSpec, spec2 *olm.CatalogSourceS
 		strings.EqualFold(spec1.Address, spec2.Address) &&
 		strings.EqualFold(spec1.DisplayName, spec2.DisplayName) &&
 		strings.EqualFold(spec1.Publisher, spec2.Publisher) &&
-		strings.EqualFold(spec1.Image, spec2.Image) &&
-		spec1.UpdateStrategy.RegistryPoll == spec1.UpdateStrategy.RegistryPoll {
+		strings.EqualFold(spec1.Image, spec2.Image) {
+		if spec1.UpdateStrategy != nil && spec2.UpdateStrategy != nil {
+			if spec1.UpdateStrategy.RegistryPoll == spec1.UpdateStrategy.RegistryPoll {
```

Comment:
I think it would be more clear if we had the polling nil check done separately from the string checks. If the strings are not equal then we can return early, otherwise we can proceed to check the polling